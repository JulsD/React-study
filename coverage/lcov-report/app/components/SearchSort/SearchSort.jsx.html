<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/SearchSort/SearchSort.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/SearchSort</a> SearchSort.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/8</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { connect } from 'react-redux'
import { setSortBy } from '../../action'
&nbsp;
import styles from './search-sort.css';
&nbsp;
export class SearchSort extends React.Component {
&nbsp;
  constructor(props) {
    super(props);
&nbsp;
    this.handleInputChange = this.handleInputChange.bind(this);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleInputChange(event) {
    let target = <span class="cstat-no" title="statement not covered" >event.target;</span>
    let value = <span class="cstat-no" title="statement not covered" >target.value;</span>
<span class="cstat-no" title="statement not covered" >    this.props.setSortBy(value);</span>
  }
&nbsp;
  render() {
    return (
      &lt;div className = {styles.root}&gt;
        &lt;h4&gt;Sort by&lt;/h4&gt;
        &lt;label className = {this.props.sortBy ==='release_date' ? <span class="branch-0 cbranch-no" title="branch not covered" >styles.active </span>: ''}&gt;
          release date
          &lt;input type='radio'
                 name='sortBy'
                 value='release_date'
                 onChange={this.handleInputChange}
                 checked={this.props.sortBy ==='release_date'}/&gt;
        &lt;/label&gt;
        &lt;label className = {this.props.sortBy ==='vote_average' ? <span class="branch-0 cbranch-no" title="branch not covered" >styles.active </span>: ''}&gt;
          rating
          &lt;input type='radio'
                 name='sortBy'
                 value='vote_average'
                 onChange={this.handleInputChange}
                 checked={this.props.sortBy ==='vote_average'} /&gt;
        &lt;/label&gt;
      &lt;/div&gt;
    );
  }
&nbsp;
}
&nbsp;
const actions = {
  setSortBy
}
&nbsp;
function <span class="fstat-no" title="function not covered" >mapProps(</span>state) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    sortBy: state.sort
  }
}
&nbsp;
export default connect(mapProps, actions)(SearchSort);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 09 2017 00:12:39 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
