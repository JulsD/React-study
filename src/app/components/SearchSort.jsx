import React from 'react';

class SearchSort extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      sortBy: 'release date'
    };

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    console.log(name + ':' + value);

    this.setState({
      [name]: value
    });
  }

  render() {
    return (
      <div className="search-sort">
        <h4>Sort by</h4>
        <label>
          release date
          <input type="radio"
                 name="sortBy"
                 value="release date"
                 onChange={this.handleInputChange}
                 checked={this.state.sortBy ==="release date"}/>
        </label>
        <label>
          rating
          <input type="radio"
                 name="sortBy"
                 value="rating"
                 onChange={this.handleInputChange}
                 checked={this.state.sortBy ==="rating"} />
        </label>
      </div>
    );
  }

}

export default SearchSort;
