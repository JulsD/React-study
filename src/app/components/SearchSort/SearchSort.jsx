import React from 'react';
import { connect } from 'react-redux'
import { setSort } from '../../action/sort'

import styles from './search-sort.css';

class SearchSort extends React.Component {

  constructor(props) {
    super(props);

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    let target = event.target;
    let value = target.value;
    this.props.setSort(value);
  }

  render() {
    return (
      <div className = {styles.root}>
        <h4>Sort by</h4>
        <label className = {this.props.sortBy ==='release_date' ? styles.active : ''}>
          release date
          <input type='radio'
                 name='sortBy'
                 value='release_date'
                 onChange={this.handleInputChange}
                 checked={this.props.sortBy ==='release_date'}/>
        </label>
        <label className = {this.props.sortBy ==='vote_average' ? styles.active : ''}>
          rating
          <input type='radio'
                 name='sortBy'
                 value='vote_average'
                 onChange={this.handleInputChange}
                 checked={this.props.sortBy ==='vote_average'} />
        </label>
      </div>
    );
  }

}

const actions = {
  setSort
}

function mapProps(state) {
  return {
    sortBy: state.sort
  }
}

export default connect(mapProps, actions)(SearchSort);
