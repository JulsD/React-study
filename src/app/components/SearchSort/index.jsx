import SearchSort from './SearchSort';
export default SearchSort;