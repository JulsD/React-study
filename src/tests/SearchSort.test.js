import React from 'react';
import { SearchSort } from '../app/components/SearchSort/SearchSort';

test('SearchSort should render a <nav>', () => {
  const testStore = {
    setSortBy: () => true,
    sortBy: 'title'
  }
  const wrapper = shallow(
    <SearchSort setSortBy={testStore.setSortBy} sortBy={testStore.sortBy}/>
  );
  expect(wrapper).toMatchSnapshot();
});
