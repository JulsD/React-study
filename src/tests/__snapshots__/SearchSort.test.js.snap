// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSort should render a <nav> 1`] = `
<div
  className="root"
>
  <h4>
    Sort by
  </h4>
  <label
    className=""
  >
    release date
    <input
      checked={false}
      name="sortBy"
      onChange={[Function]}
      type="radio"
      value="release_date"
    />
  </label>
  <label
    className=""
  >
    rating
    <input
      checked={false}
      name="sortBy"
      onChange={[Function]}
      type="radio"
      value="vote_average"
    />
  </label>
</div>
`;
